src/openrct2-ui/windows/RideConstruction.cpp

```diff
@@ -3561,9 +3560,12 @@ void ride_construction_toolupdate_construct(ScreenCoordsXY screenCoords)
     }
 
     _previousTrackPiece = _currentTrackBegin;
+    // search for appropriate z value for ghost, up to max ride height
+    int numAttempts = (z <= (8 * MAX_TRACK_HEIGHT) ? MAX_TRACK_HEIGHT - (z / 8) + 1 : 2);
```

Comment:
This is hard to read and uses hard coded `8`s instead of constants.

I'd prefer the following:
```suggestion
    const auto smallZ = z / COORDS_Z_STEP;
    int numAttempts = (smallZ <= MAX_TRACK_HEIGHT ? (MAX_TRACK_HEIGHT - smallZ + 1) : 2);
```